_posts/2019-09-11-how-to-stay-up-to-date.markdown

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: How to stay up to date with your Rails application
+date: 2019-08-27 13:56:00
+categories: ["Rails", "Upgrade"]
+author: bronzdoc
+---
+
+How to stay up to date with your rails application
+
+<!--more-->
+
+In Fastruby.io we work with alot of clients which have outdated Rails applications, we help them upgrade to a newer Rails version.
+
+An outdated Rals application doesn't happen over night, and there are things that you could start doing to avoid going into an outdated project, that will be difficult to maintain and migrate later on.
+
+### Stop monkey patching
+
+In our experience having monkey patched gems when doing an upgrade usually is one of the hardest things to deal with since work hours would have to be put to make the monkey patched gem compatible with the new Rails version. So Have that in mind before dedicating time to monkey patch rails core libraries or gems that depend on an specific rails versions.
+
+### Start treating Deprecations warnings as errors.
+
+In Rails you can configure this option, so everytime a deprecation warning occurs you can address it as soon as you can.
+This will help you to fix the issue rigth a away and be prepared for the next version jump.
+
+```ruby
+  # Raise error on deprecation¬
+  config.active_support.deprecation = :raise
+```
+
+### Check for outdated gems
+
+You can use bundle outdated to check outdated gems in your current bundle. While `bundle outdated` is quite useful it doesn't give us details relative to our own `Gemfile`. In Ombulabs we like to use  [next_rails](https://rubygems.org/gems/next_rails/versions/1.0.0) this gem will give us a set of tools to check for outdated gems, check incompatibilities and even bootstrap an upgrade project.
```

Comment:
@bronzdoc It would be great if you can expand on this. It would be great if you can write a rake task that checks for `bundle outdated` and if it finds any _new_ outdated dependencies it exits with a non-zero status. It is similar to the idea of running `bundler audit` and failing the test suite if there is a new known vulnerability: https://github.com/alphagov/e-petitions/blob/master/Rakefile#L11-L14

Reply:
Well i think something is wrong here, i think `bundle outdated` do give us details relative to our Gemfile, so i'll have to rephrase this i think.